### ISSUE TITLE
[Write | EmailConfirmation] create slice ApproveEmailConfirmation -> EmailConfirmationWasApproved

### ISSUE THREAD

**Issue by nowakprojects** [CodersCrew/coderscamp#284]
**Dependencies (blockers)**
- [x] https://github.com/CodersCrew/coderscamp/issues/285
- [x] Make sure that #279 is done (must be merged before)

**Subtasks**
If you need more info or have questions, fell free to ask @nowakprojects.

Command definition:
- [x] Create command type `ApproveEmailConfirmation` in `modules/shared/commands`
- [x] Create `ApproveEmailConfirmationApplicationCommand` class in `modules/shared/commands`

Domain logic:
- [x] Create function approveEmailConfirmation in `modules/write/email-confirmation/domain`.
- [x] Return EmailConfirmationWasApproved from the method, only if correspondent (last not approved request, need to have same confirmationToken) EmailConfirmationWasRequested lastly and was not already approved. Read it from events with .reduce() method.

Application logic:
- [x] Create CommandHandler for ApproveEmailConfirmationApplicationCommand
- [x] Execute domain logic. Event stream name should be in format `EmailConfirmation_${userId}_${confirmationFor}` _Do you know why? Try to elaborate before taking this task._

Presentation logic:
- Rest API endpoint will be done in another task.

NestJS Module:
- [x] Your work should be done in EmailConfirmationWriteModule.

**Event Modeling**
MIRO link: https://miro.com/app/board/o9J_lQvnN28=/?moveToWidget=3074457363395418452&cot=14
